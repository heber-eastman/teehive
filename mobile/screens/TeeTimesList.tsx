import React from 'react';
import { View, Text, FlatList, StyleSheet, TouchableOpacity, Linking } from 'react-native';
import { useTeeTimes, TeeTime } from '../hooks/useTeeTimes';
import TeeTimeCard from '../components/TeeTimeCard';

export default function TeeTimesList() {
  const { teeTimes, isLoading, error, refetch } = useTeeTimes();



  const handleBooking = (url: string) => {
    Linking.openURL(url);
  };

  const renderTeeTime = ({ item }: { item: TeeTime }) => {
    return (
      <TeeTimeCard 
        teeTime={item}
        onPress={() => handleBooking(item.bookingUrl)}
      />
    );
  };

  if (isLoading) {
    return (
      <View style={styles.centerContainer}>
        <Text style={styles.loadingText}>Loading tee times...</Text>
      </View>
    );
  }

  if (error) {
    return (
      <View style={styles.centerContainer}>
        <Text style={styles.errorText}>Error: {error.message}</Text>
        <TouchableOpacity style={styles.retryButton} onPress={refetch}>
          <Text style={styles.retryText}>Retry</Text>
        </TouchableOpacity>
      </View>
    );
  }

  return (
    <View style={styles.container}>
      <FlatList
        data={teeTimes}
        renderItem={renderTeeTime}
        keyExtractor={(item) => item.id}
        contentContainerStyle={styles.listContainer}
        showsVerticalScrollIndicator={false}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f5f5f5',
  },
  listContainer: {
    padding: 16,
  },
  centerContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
  },
  loadingText: {
    fontFamily: 'System',
    fontSize: 16,
    color: '#666',
  },
  errorText: {
    fontFamily: 'System',
    fontSize: 16,
    color: '#d32f2f',
    textAlign: 'center',
    marginBottom: 16,
  },
  retryButton: {
    backgroundColor: '#1976d2',
    paddingHorizontal: 20,
    paddingVertical: 10,
    borderRadius: 6,
  },
  retryText: {
    fontFamily: 'System',
    fontSize: 14,
    color: '#ffffff',
    fontWeight: '500',
  },
}); 