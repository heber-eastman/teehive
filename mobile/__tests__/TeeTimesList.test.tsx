import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider as PaperProvider } from 'react-native-paper';
import TeeTimesList from '../screens/TeeTimesList';

// Mock the theme
jest.mock('react-native-paper', () => {
  const original = jest.requireActual('react-native-paper');
  return {
    ...original,
    useTheme: () => ({
      colors: {
        primary: '#000000',
      },
    }),
  };
});

const mockTeeTimes = [
  {
    id: '1',
    courseName: 'Pine Valley Golf Club',
    dateTime: '2024-03-20T10:00:00Z',
    spotsAvailable: 4,
    holes: 18,
    priceAmount: 150,
    currency: 'USD',
    bookingUrl: 'https://example.com/book1',
    createdAt: '2024-03-19T00:00:00Z',
    updatedAt: '2024-03-19T00:00:00Z'
  },
  {
    id: '2',
    courseName: 'Augusta National',
    dateTime: '2024-03-21T14:30:00Z',
    spotsAvailable: 2,
    holes: 18,
    priceAmount: 200,
    currency: 'USD',
    bookingUrl: 'https://example.com/book2',
    createdAt: '2024-03-19T00:00:00Z',
    updatedAt: '2024-03-19T00:00:00Z'
  },
  {
    id: '3',
    courseName: 'St Andrews Links',
    dateTime: '2024-03-22T09:15:00Z',
    spotsAvailable: 1,
    holes: 9,
    priceAmount: 175,
    currency: 'GBP',
    bookingUrl: 'https://example.com/book3',
    createdAt: '2024-03-19T00:00:00Z',
    updatedAt: '2024-03-19T00:00:00Z'
  }
];

describe('TeeTimesList', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <PaperProvider>
        <TeeTimesList teeTimes={mockTeeTimes} onRefresh={() => {}} />
      </PaperProvider>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders all mock tee times', () => {
    const { getByText } = render(
      <PaperProvider>
        <TeeTimesList teeTimes={mockTeeTimes} onRefresh={() => {}} />
      </PaperProvider>
    );

    // Check if all course names are rendered
    expect(getByText('Pine Valley Golf Club')).toBeTruthy();
    expect(getByText('Augusta National')).toBeTruthy();
    expect(getByText('St Andrews Links')).toBeTruthy();

    // Check if spots and holes are rendered
    expect(getByText(/4 spots • 18 holes/)).toBeTruthy();
    expect(getByText(/2 spots • 18 holes/)).toBeTruthy();
    expect(getByText(/1 spots • 9 holes/)).toBeTruthy();

    // Check if prices are rendered
    expect(getByText('$150.00')).toBeTruthy();
    expect(getByText('$200.00')).toBeTruthy();
    expect(getByText('£175.00')).toBeTruthy();
  });
}); 