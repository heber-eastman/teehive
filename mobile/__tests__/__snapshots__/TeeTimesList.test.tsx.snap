// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeeTimesList renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "flexGrow": 1,
          "padding": 16,
        }
      }
      data={
        [
          {
            "bookingUrl": "https://example.com/book1",
            "courseName": "Pine Valley Golf Club",
            "currency": "USD",
            "dateTime": "2024-03-20T10:00:00Z",
            "holes": 18,
            "id": "1",
            "priceAmount": 150,
            "spotsAvailable": 4,
          },
          {
            "bookingUrl": "https://example.com/book2",
            "courseName": "Augusta National",
            "currency": "USD",
            "dateTime": "2024-03-21T14:30:00Z",
            "holes": 18,
            "id": "2",
            "priceAmount": 200,
            "spotsAvailable": 2,
          },
          {
            "bookingUrl": "https://example.com/book3",
            "courseName": "St Andrews Links",
            "currency": "GBP",
            "dateTime": "2024-03-22T09:15:00Z",
            "holes": 9,
            "id": "3",
            "priceAmount": 175,
            "spotsAvailable": 1,
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                false,
                [
                  {
                    "paddingRight": 24,
                    "paddingVertical": 8,
                  },
                  {
                    "backgroundColor": "white",
                    "borderRadius": 8,
                    "elevation": 2,
                    "marginBottom": 8,
                  },
                ],
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                    "marginVertical": 6,
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  [
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    {
                      "alignSelf": "center",
                      "marginLeft": 16,
                      "marginRight": 0,
                    },
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "color": "rgba(73, 69, 79, 1)",
                        "fontSize": 24,
                      },
                    ]
                  }
                >
                  □
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "paddingLeft": 16,
                    },
                    {
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                testID="undefined-content"
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                      },
                      {
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontSize": 16,
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Pine Valley Golf Club
                </Text>
                <Text
                  numberOfLines={2}
                  onTextLayout={[Function]}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                      },
                      {
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontSize": 14,
                        },
                        {
                          "color": "rgba(73, 69, 79, 1)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  4 spots • 18 holes
                </Text>
              </View>
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "alignSelf": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                      },
                      {
                        "color": "#000000",
                      },
                    ],
                  ]
                }
              >
                $150.00
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                false,
                [
                  {
                    "paddingRight": 24,
                    "paddingVertical": 8,
                  },
                  {
                    "backgroundColor": "white",
                    "borderRadius": 8,
                    "elevation": 2,
                    "marginBottom": 8,
                  },
                ],
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                    "marginVertical": 6,
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  [
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    {
                      "alignSelf": "center",
                      "marginLeft": 16,
                      "marginRight": 0,
                    },
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "color": "rgba(73, 69, 79, 1)",
                        "fontSize": 24,
                      },
                    ]
                  }
                >
                  □
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "paddingLeft": 16,
                    },
                    {
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                testID="undefined-content"
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                      },
                      {
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontSize": 16,
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Augusta National
                </Text>
                <Text
                  numberOfLines={2}
                  onTextLayout={[Function]}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                      },
                      {
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontSize": 14,
                        },
                        {
                          "color": "rgba(73, 69, 79, 1)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  2 spots • 18 holes
                </Text>
              </View>
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "alignSelf": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                      },
                      {
                        "color": "#000000",
                      },
                    ],
                  ]
                }
              >
                $200.00
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                false,
                [
                  {
                    "paddingRight": 24,
                    "paddingVertical": 8,
                  },
                  {
                    "backgroundColor": "white",
                    "borderRadius": 8,
                    "elevation": 2,
                    "marginBottom": 8,
                  },
                ],
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                    "marginVertical": 6,
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  [
                    {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    {
                      "alignSelf": "center",
                      "marginLeft": 16,
                      "marginRight": 0,
                    },
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "color": "rgba(73, 69, 79, 1)",
                        "fontSize": 24,
                      },
                    ]
                  }
                >
                  □
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "paddingLeft": 16,
                    },
                    {
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                testID="undefined-content"
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                      },
                      {
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontSize": 16,
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  St Andrews Links
                </Text>
                <Text
                  numberOfLines={2}
                  onTextLayout={[Function]}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                      },
                      {
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontSize": 14,
                        },
                        {
                          "color": "rgba(73, 69, 79, 1)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  1 spots • 9 holes
                </Text>
              </View>
              <Text
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "alignSelf": "center",
                        "fontSize": 16,
                        "fontWeight": "bold",
                      },
                      {
                        "color": "#000000",
                      },
                    ],
                  ]
                }
              >
                £175.00
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;
